cloudformation/ad/ad-integration.yaml

```diff
@@ -380,13 +395,28 @@ Resources:
               sleep 0.5
               echo "Registering User..."
               echo $ADMIN_PW | adcli create-user -x -U Admin --domain=${DirectoryDomain} --display-name=${UserName} ${UserName}
+
+              echo "Creating domain certificate..."
+              PRIVATE_KEY="${DirectoryDomain}.key"
+              CERTIFICATE="${DirectoryDomain}.crt"
+              printf ".\n.\n.\n.\n.\n${DirectoryDomain}\n.\n" | openssl req -x509 -sha256 -nodes -newkey rsa:2048 -keyout $PRIVATE_KEY -days 90 -out $CERTIFICATE
```

Comment:
> where is this key generated? is it generated with the right permissions?

Both the certificate and the private key are generated locally to the instance, but they are not meant to be used there. In fact, they are only meant to be uploaded to secrets manager. So, permissions is not an issue and both files can be deleted once uploaded to Secrets Manager.

> do we need to keep the key locally or we can delete it once we upload it to secrets manager?

We can delete both certificate and private key from the instance. Good point, will fix in the next rev.

Reply:
> also if not needed on this instance have we considered generating this directly in Lambda?

Generating certificate and private key within Lambda was actually the first approach that I considered. At the end I discarded it because it would have required to include external dependencies in the Lambda (pyOpenSSL). So, doing that within a Lambda would have introduced more complexity to the template (Lambda code on S3, Lambda Layers, ...)

Reply:
>Generating certificate and private key within Lambda was actually the first approach that I considered. At the end I discarded it because it would have required to include external dependencies in the Lambda (pyOpenSSL). So, doing that within a Lambda would have introduced more complexity to the template (Lambda code on S3, Lambda Layers, ...)

makes sense, for some reason I was assuming that openssl is available in the lambda default runtime and you could just use it.